pywhy_stats/independence.py

```diff
@@ -44,5 +53,30 @@ def independence_test(
     result : PValueResult
         An instance of the PValueResult data class, containing the p-value, test statistic,
         and any additional information related to the independence test.
+
+    See Also
+    --------
+    fisherz : Fisher's Z test for independence
     """
-    pass
+    method_func: ModuleType
+    if method == Methods.AUTO:
+        method_func = Methods.FISHERZ
+    else:
+        method_func = method
+
+    if method_func == Methods.FISHERZ:
+        if condition_on is None:
+            data = [X, Y]
+        else:
+            data = [X, Y, condition_on]
+        for _data in data:
+            _, pval = scipy.stats.normaltest(_data)
+
+            # XXX: we should add pinguoin as an optional dependency for doing multi-comp stuff
+            if pval < 0.05:
+                warn("Your data is not all normal but you're trying to use partial correlation")
+
+    if condition_on is None:
+        return method_func.ind(X, Y, method, **kwargs)
```

Comment:
Would this imply that `method_func` is a module? Not sure if something like this is common. What about we only accept an enum as input here? I don't see a use case where someone creates their own module and then wants to call his helper function with it.

Reply:
I renamed it to `method_module`